[delonlee01] iP

delonlee01 opened the pull request:
# Woody 🤠 

> A cowboy without his hat is like a, yodel without an ay-ee-hoo! ([source](https://toystorymerchandise.fandom.com/wiki/Playtime_Sheriff_Woody))

## Table of Contents
- [Overview](#overview)
- [Features](#features)
- [Commands](#commands)
- [For Developers](#for-developers)

## Overview

**Woody** is a chatbot assistant inspired by the character in the _Toy Story_ movie.

## Features

- [X] Managing tasks
- [X] Managing deadlines
- [X] Managing events
- [ ] and more... TBC

## Commands[^1]

1. `list` - prints all tasks
2. `check <date>` - prints all tasks related to the date
3. `find <keyword>` - prints all tasks with the keyword in its description
4. `todo <description>` - adds a todo task
5. `deadline <description> /by <date>` - adds a deadline task
6. `event <description> /from <date> /to <date>` - adds a deadline task
7. `delete <index>` - deletes a task at the specified index
8. `mark <index>` - marks a task at the specified index as done
9. `unmark <index>` - marks a task at the specified index as not done
10. `bye` - ends the session

## For Developers

The abstract class that acts as a skeleton for each command:

```java
public abstract class Command {
    public abstract void execute(TaskList tasks, Ui ui) throws WoodyException;

    public boolean isReadOnly() {
        return false;
    }

    public boolean isExit() {
        return false;
    }
}
```

[^1]: Date format `dd/MM/yyyy` (e.g. `01/01/2025`)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

delonlee01 committed:
Rename chatbot to woody

delonlee01 committed:
Update greeting and exit message

delonlee01 committed:
Remove unwanted traces of Duke

delonlee01 committed:
Add user input support

delonlee01 committed:
Customize greeting and exit message

delonlee01 committed:
Add list command

delonlee01 committed:
Add mark/unmark task feature

delonlee01 committed:
Replace Pair class with Task class

delonlee01 committed:
Add Todo, Event, Deadline

delonlee01 committed:
Update text UI test

delonlee01 committed:
Remove redundant codes

delonlee01 committed:
Add exceptions to handle errors

delonlee01 committed:
Update text UI test

delonlee01 committed:
Add delete command

delonlee01 committed:
Fix IndexOutOfBoundsException error

delonlee01 committed:
Move related files into packages

delonlee01 committed:
Add TaskList

delonlee01 committed:
Modify TaskList to read and write data to file

delonlee01 committed:
Add new text UI test

delonlee01 committed:
Fix NoSuchElementException in text UI test

delonlee01 committed:
Merge branch 'branch-Level-7'

delonlee01 committed:
Modify tasks to store date strings as LocalDate

delonlee01 committed:
Update text UI tests

delonlee01 committed:
Add check command

delonlee01 committed:
Merge branch 'branch-Level-8'

delonlee01 committed:
Add Command, Parser, and Ui class

delonlee01 committed:
Add Storage class

delonlee01 committed:
Fix text UI tests

delonlee01 committed:
Merge branch 'branch-A-MoreOOP'

delonlee01 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

delonlee01 committed:
Update build.gradle to work with woody

delonlee01 committed:
Add checkstyle for gradle

delonlee01 committed:
Merge branch 'branch-A-Gradle'

delonlee01 committed:
Add JUnit tests for task package

delonlee01 committed:
Add JUnit tests for command package

delonlee01 committed:
Delete unused method in ByeCommand

delonlee01 committed:
Merge branch 'branch-A-JUnit'

delonlee01 committed:
Add JavaDoc comments for task package

delonlee01 committed:
Add JavaDoc comments for exception package

delonlee01 committed:
Add JavaDoc comments for command package

delonlee01 committed:
Add JavaDoc comments for woody package

delonlee01 committed:
Update formatting of JavaDoc comments

delonlee01 committed:
Update code style

delonlee01 committed:
Fix checkstyle warnings

delonlee01 committed:
Add find command

delonlee01 committed:
Update text UI tests

delonlee01 committed:
Add FindCommandTest

delonlee01 committed:
Merge branch 'branch-A-JavaDoc'

delonlee01 committed:
Merge branch 'master' into branch-A-CodingStandard

delonlee01 committed:
Fix checkstyle and JavaDoc comments

delonlee01 committed:
Merge branch 'branch-A-CodingStandard'

delonlee01 committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/task/Task.java
#	src/main/java/woody/Parser.java

delonlee01 committed:
Add JavaDoc comments

delonlee01 committed:
Fix checkstyle

delonlee01 committed:
Merge branch 'branch-Level-9'

delonlee01 committed:
Update JavaDoc comments

delonlee01 committed:
Merge branch 'branch-A-CheckStyle'

philbertshea commented on src/main/java/command/TodoCommand.java:
```diff
@@ -0,0 +1,49 @@
+     * @return TodoCommand
+     */
+    public static TodoCommand createCommandIfValid(String input) {
+        Matcher matcher = REGEX_PATTERN.matcher(input);
```
Maybe you can consider renaming this to inputMatcher or inputRegexMatcher for clarity, but this looks fine to me as well.

philbertshea commented on src/main/java/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,56 @@
+
+    /**
+     * Returns a DeadlineCommand if the specified input matches the
+     * usage format.
```
Well-written,. Could you add ", else return null."?

philbertshea commented on src/main/java/task/Todo.java:
```diff
@@ -0,0 +1,27 @@
+package task;
+
+/**
+ * Represents a Todo task in the chatbot system. A
```
Just for curiosity, is there any reason why "A" is separated from the rest of the second sentence? Haha

philbertshea commented on src/main/java/woody/Parser.java:
```diff
@@ -0,0 +1,74 @@
+        Object[] args = new Object[]{text};
+        for (var commandClass : commands) {
+            try {
+                Command command = (Command) commandClass.getMethod("createCommandIfValid", params).invoke(null, args);
```
Is it possible to use line wrapping for this line? E.g. break the .invoke call to line 41. Looks a bit long, not sure if it exceeds 120 chars. If it doesn't then it's fine, thanks.

philbertshea commented on src/main/java/woody/Ui.java:
```diff
@@ -0,0 +1,84 @@
+ * system.
+ */
+public class Ui {
+    private static final String LOGO = """
```
Not related to coding standards, but cool logo you got there!

philbertshea commented on src/main/java/woody/Ui.java:
```diff
@@ -0,0 +1,84 @@
+     * Sets the suppression status of the user interface. If true, outputs are not
+     * printed.
+     *
+     * @return Ui
```
Maybe add a description of the returned Ui object?

philbertshea commented on src/main/java/woody/Ui.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    /**
+     * Prints output to standard out.
```
I guess you could add the parameter String text in the comment for this method?

philbertshea reviewed (commented):
Looks great to me! 🥇  Conscientious following of Java coding standards (definitely more conscientious than me haha). A few suggestions and keep up the fantastic work.

delonlee01 committed:
Add the basic GUI

delonlee01 committed:
Update GUI to use FXML

delonlee01 committed:
Update woody to use GUI

delonlee01 committed:
Update tests

delonlee01 committed:
Add CSS to FXML

delonlee01 committed:
Merge branch 'branch-Level-10'

Zerolegacy commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+ * contains a description, done status, and 'by' date.
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```
Perhaps the variable name can be changed to another name that has a clearer explanation?
```suggestion
    protected LocalDate dueByDate;
```

Zerolegacy commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,59 @@
+ * description, done status, 'from' date, and 'to' date.
+ */
+public class Event extends Task {
+    protected LocalDate from;
+    protected LocalDate to;
```
Perhaps the variable names can be changed to something that has a clearer explanation?
```suggestion
    protected LocalDate fromDate;
    protected LocalDate toDate;
```

Zerolegacy commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+    /**
+     * Adds a new task to the TaskList.
+     *
+     * @param task Task
```
Should the @param line be omitted?

Zerolegacy commented on src/test/java/command/CheckCommandTest.java:
```diff
@@ -0,0 +1,56 @@
+import task.TaskList;
+
+public class CheckCommandTest {
+    private static final LocalDate DATE_1 = LocalDate.parse("01/01/2025",
+            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+    private static final LocalDate DATE_2 = LocalDate.parse("02/01/2025",
+            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
+    private static final LocalDate DATE_3 = LocalDate.parse("03/01/2025",
+            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
```
Perhaps the numbers in the variable names can be changed to something that has a clearer explanation. The Coding quality guide indicates not use numbers in our variable names but I am also not too sure if it should apply in this case.
```suggestion
    private static final LocalDate DATE_FIRST_JAN = LocalDate.parse("01/01/2025",
            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
    private static final LocalDate DATE_SECOND_JAN = LocalDate.parse("02/01/2025",
            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
    private static final LocalDate DATE_THIRD_JAN = LocalDate.parse("03/01/2025",
            DateTimeFormatter.ofPattern("dd/MM/yyyy"));
```

Zerolegacy commented on src/main/java/woody/Storage.java:
```diff
@@ -0,0 +1,97 @@
+    /**
+     * Stores all tasks in the specified TaskList object locally.
+     *
+     * @param taskList TaskList
```
Should the @param line be omitted?

Zerolegacy reviewed (commented):
LGTM! Overall nicely written code. It was clean and easy to read. No coding standard violations and minimal coding quality violations. Good work!! :)

delonlee01 committed:
Enable assertions

delonlee01 committed:
Add assertions

delonlee01 committed:
Refactor guard clauses

Refactor guard clauses

Some guard clauses implemented handles the happy path instead of the 'unusual' cases. Refactored these guard clauses to make the happy path prominent.

delonlee01 committed:
Refactor code structure

delonlee01 committed:
Update comments

delonlee01 committed:
Merge pull request #1 from delonlee01/branch-A-Assertions

Add assertions

delonlee01 committed:
Merge pull request #2 from delonlee01/branch-A-CodeQuality

Update code quality

delonlee01 committed:
Add Github workflow config file

delonlee01 committed:
Merge branch 'branch-A-CI'

delonlee01 committed:
Add tag command

Add tag command

To support tagging of tasks, added a new command to handle achieve it.

delonlee01 committed:
Update Storage class to save and load tags

Update Storage class to save and load tags

The local storage implementation does not include saving and loading of tags. Updated the implementation to do so.

delonlee01 committed:
Update test cases

delonlee01 committed:
Merge branch 'BCD-Extension'

delonlee01 committed:
Fix ArrayIndexOutOfBoundsException on data load

delonlee01 committed:
Update return message of mark and unmark commands

delonlee01 committed:
Merge branch 'branch-A-MoreErrorHandling'

delonlee01 committed:
Add credits

delonlee01 committed:
Update user guide

delonlee01 committed:
Merge branch 'branch-A-UserGuide'

delonlee01 committed:
Update screenshot of product

delonlee01 committed:
Merge branch 'branch-A-UserGuide'
